pilot/pkg/networking/core/cluster.go

```diff
@@ -653,7 +653,7 @@ func buildInboundClustersFromSidecar(cb *ClusterBuilder, proxy *model.Proxy,
 				TargetPort:  uint32(port),
 			},
 		}
-		localCluster := cb.buildInboundCluster(int(ingressListener.Port.Number), endpointAddress, proxy, endpoint, nil)
```

Comment:
I think this is not equivilent?

Before: addTelemetryMetadata is called with inboundServices=nil

After:  addTelemetryMetadata is called with inboundServices=`[endpoint]`

Reply:
It should be included in the telemetry

Reply:
why? This is a fake synthetic endpoint for Sidecar ingress case. 

The PR is indicated as a refactoring and is introducing a behavioral change. If the behavioral change is intended can you explain the motivation and the difference, from user perspective, before and after? 

Right now it seems accidental

Reply:
Because we have used that in listener builder as well. I didnot know the diff behavior with listener before, but via this pr. I can see that.

```
		cc := inboundChainConfig{
				// Sidecar config doesn't have a real hostname. In order to give some telemetry info, make a synthetic hostname.
				telemetryMetadata: telemetry.FilterChainMetadata{
					InstanceHostname: host.Name(lb.node.SidecarScope.Name + "." + lb.node.SidecarScope.Namespace),
				},
```

Reply:
that is used for a different purpose though? 

Can you explain from a user standpoint what the change is here? Without context it seems like we are just accidentally introducing behavioral changes during a refactoring so I want to better understand the goal.

Reply:
This cluster metadata "services" is used by istio stats, https://github.com/istio/proxy/blob/50d48187043c6f96b6b7687930e03496b3df6cb8/source/extensions/filters/http/istio_stats/istio_stats.cc#L1039

We have generated metadata for real service instances. And here
only if there is a Sidecar.Ingress has no associated service, we will generate with hostname with sidecarname + ns.

Reply:
So , my understanding is this change makes it so we will report in the telemetry this fake hostname where before we would not. I don't understand why we would want that?

Reply:
Just Like in the listener and also route, it is just an indication, no other function

Reply:
what do you mean just an indicator? it ends up it the final user-facing telemetry metrics. So with this change, we are going to be putting made up hostnames into the Istio metrics.

Reply:
and have done for inbound listener and route, so  why not

Reply:
those do not show up in the istio standard telemetry

Reply:
Any diff in envoy or istio, to users they are all managed by istio

Reply:
Istio telemetry is a core part of istio with stable API since v1.0. Envoy telemetry is break glass, undocumented, advanced feature  without stability guarantees

Reply:
So If the sidecar ingress has associated service, we did miss telemetry info. What would you say about that?

FYI: `findOrCreateServices()`, first matches with real service targets, if not ffound, construct with sidecar

Reply:
I am confused if we should keep this ambiguity

Reply:
I am fine including it if it's a real service probably, but not the fake one which is why the code is structured the way it is